jdbc/src/main/java/com/interface21/jdbc/core/JdbcTemplate.java

```diff
@@ -22,58 +21,64 @@ public JdbcTemplate(final DataSource dataSource) {
         this.dataSource = dataSource;
     }
 
-    public void update(final String sql, final Object... args) {
-        try (final Connection conn = dataSource.getConnection();
-             final PreparedStatement pstmt = conn.prepareStatement(sql)) {
+    public int update(final String sql, final Object... args) {
+        return update(sql, createPreparedStatementSetter(args));
+    }
 
-            log.debug("query : {}", sql);
-            setParameters(pstmt, args);
-            pstmt.executeUpdate();
-        } catch (SQLException e) {
-            log.error(e.getMessage(), e);
-            throw new DataAccessException(e);
-        }
+    public int update(final String sql, final PreparedStatementSetter pss) {
+        return execute(sql, pstmt -> {
+            pss.setValues(pstmt);
+            return pstmt.executeUpdate();
+        });
     }
 
-    public <T> T findOne(final String sql, final RowMapper<T> rowMapper, final Object... args) {
-        try (final Connection conn = dataSource.getConnection();
-             final PreparedStatement pstmt = conn.prepareStatement(sql)) {
+    public <T> T queryForObject(final String sql, final RowMapper<T> rowMapper, Object... args) {
+        return queryForObject(sql, createPreparedStatementSetter(args), rowMapper);
+    }
 
-            log.debug("query : {}", sql);
-            setParameters(pstmt, args);
+    public <T> T queryForObject(final String sql, final PreparedStatementSetter pss, final RowMapper<T> rowMapper) {
+        final List<T> results = query(sql, pss, rowMapper);
+        if (results.isEmpty()) {
+            throw new DataAccessException("Incorrect result size: expected 1, actual 0");
+        }
+        return results.getFirst();
+    }
 
-            try (final ResultSet rs = pstmt.executeQuery()) {
-                if (rs.next()) {
-                    return rowMapper.mapRow(rs);
+    public <T> List<T> query(final String sql, final RowMapper<T> rowMapper, Object... args) {
+        return query(sql, createPreparedStatementSetter(args), rowMapper);
+    }
+
+    public <T> List<T> query(final String sql, final PreparedStatementSetter pss, final RowMapper<T> rowMapper) {
+        return execute(sql, pstmt -> {
+            pss.setValues(pstmt);
+            try (ResultSet rs = pstmt.executeQuery()) {
```

Comment:
여기에도 일괄적으로 `final` 붙여주면 좋을 것 같아요!

```suggestion
            try (final ResultSet rs = pstmt.executeQuery()) {
```